compiler/rustc_monomorphize/src/partitioning/default.rs

```diff
@@ -278,7 +278,8 @@ fn characteristic_def_id_of_mono_item<'tcx>(
                 | ty::InstanceDef::Intrinsic(..)
                 | ty::InstanceDef::DropGlue(..)
                 | ty::InstanceDef::Virtual(..)
-                | ty::InstanceDef::CloneShim(..) => return None,
+                | ty::InstanceDef::CloneShim(..)
+                | ty::InstanceDef::AsRepr(..) => return None,
```

Comment:
Should this return the def id of the enum it is for?

Reply:
I'm not at all familiar with this area of code, but I was assuming `AsRepr` should be treated the same as `CloneShim` here... But from my sparse understanding, both seem like reasonable implementations, so I've changed it to the enum def id